src/main/java/com/programmers/ticketparis/service/performance/PerformanceService.java

```diff
@@ -0,0 +1,64 @@
+package com.programmers.ticketparis.service.performance;
+
+import com.programmers.ticketparis.domain.performance.Performance;
+import com.programmers.ticketparis.dto.performance.request.PerformanceCreateRequest;
+import com.programmers.ticketparis.dto.performance.request.PerformanceUpdateRequest;
+import com.programmers.ticketparis.dto.performance.response.PerformanceResponse;
+import com.programmers.ticketparis.repository.performance.PerformanceRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class PerformanceService {
+    private final PerformanceRepository performanceRepository;
+
+    @Transactional
+    public void createPerformance(PerformanceCreateRequest createRequest) {
+        if (createRequest.getStartDate().isAfter(createRequest.getEndDate())) {
+            throw new IllegalArgumentException("공연 시작 날짜는 공연 종료 날짜 이후가 될 수 없습니다.");
+        }
+
+        Performance performance = createRequest.toEntity();
+        performanceRepository.save(performance);
+    }
+
+    @Transactional
+    public PerformanceResponse updatePerformance(Long id, PerformanceUpdateRequest updateRequest) {
+        if (updateRequest.getStartDate().isAfter(updateRequest.getEndDate())) {
+            throw new IllegalArgumentException("공연 시작 날짜는 공연 종료 날짜 이후가 될 수 없습니다.");
+        }
+        performanceRepository.update(id, updateRequest);
+        return findPerformanceById(id)
+                .orElseThrow(() -> new NoSuchElementException("수정할 공연 정보를 찾지 못했습니다."));
+    }
+
+    public Optional<PerformanceResponse> findPerformanceById(Long id) {
+        Performance performance = performanceRepository.findById(id)
+                .orElseThrow(() -> new NoSuchElementException("해당 공연 정보는 존재하지 않습니다."));
+
+        return Optional.ofNullable(PerformanceResponse.fromEntity(performance));
```

Comment:
`p1`

아까 위에서 언급드린 것 처럼 이미 아래 코드에서 없으면 예외를 던지고 값이 있으면 데이터를 넣어주기 때문에 아래에서 Optional.ofNullable을 사용하여 감싸줄 필요가 없다고 생각합니다!

```
Performance performance = performanceRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("해당 공연 정보는 존재하지 않습니다."));
```

```suggestion
        return PerformanceResponse.fromEntity(performance);
```

Reply:
넵! 해당 findByID에서 Exception 예외를 별도로 두면, Optional로 묶을 필요가 없어서 해당 로직을 모두 수정하였습니다.
반영: f64cdb380e287ee512801b9f547fc834fa48b4ee